### ISSUE TITLE
Go Automation API - Print JSON output during update/destroy?

### ISSUE THREAD

**Issue by luckydev** [pulumi/pulumi#18088]
I'm trying to print json output during "update" operation in Pulumi. And I'm using Pulumi automation APIs to do the same. I don't see an option here to print json output here - https://pkg.go.dev/github.com/pulumi/pulumi/sdk/v3/go/auto/optup . Am I missing anything?

I mean, I'm looking for an equivalent of `pulumi up -j` in `auto.Stack.Up()` call.

**Comment by luckydev** [comment]
I just came across this issue now - https://github.com/pulumi/pulumi/issues/7773

Should I convert this (my issue here 18088) as a feature request?

**Comment by tgummerer** [comment]
I don't think you are missing anything here, it looks like this option has not been implemented yet from a quick look at the code.  I'm marking this as an enhancement request for us to keep in mind. Thanks for opening the issue!

**Comment by jaxxstorm** [comment]
You can already do this:

```go
func processEvents(logger *zap.Logger, eventChannel <-chan events.EngineEvent) {
	for event := range eventChannel {
		jsonData, err := json.Marshal(event)
		if err != nil {
			logger.Error("Failed to marshal event to JSON", zap.Error(err))
			continue
		}
		logger.Info(string(jsonData))
	}
}
```

See here:

https://github.com/jaxxstorm/pedloy/blob/main/pkg/auto/pulumi.go
